agent/lib/kontena/service_pods/lifecycle_hook_manager.rb

```diff
@@ -0,0 +1,93 @@
+module Kontena
+  module ServicePods
+    class LifecycleHookManager
+      include Kontena::Helpers::RpcHelper
+      include Common
+
+      attr_reader :service_pod
+
+      # @param [ServicePod] service_pod
+      def initialize(service_pod)
+        @service_pod = service_pod
+      end
+
+      # @return [Boolean]
+      def on_pre_start
+        pre_start_hooks = hooks_for('pre_start')
+        return if pre_start_hooks.size == 0
+
+        service_config = config_container(service_pod.dup)
+        service_config['Entrypoint'] = '/bin/sh'
+        service_config['Cmd'] = ['-c', 'sleep 600']
```

Comment:
:+1: for also fixing the `post_start` hooks to be run with weave attached, though.

Worstcase is if the weave attach tends to win the race and users write hooks that assume that weave is attached... and they work until the timing subtly changes and they start breaking spontaneously.